qcodes/instrument_drivers/Galil/dmc_41x3.py

```diff
@@ -0,0 +1,964 @@
+
+
+class Arm:
+    """
+    Module to control probe arm
+    """
+
+    def __init__(self, controller: DMC4133Controller) -> None:
+        """
+        Initializes the arm class
+
+        Args:
+            controller: an instance of DMC4133Controller
+        """
+        self.controller = controller
+
+        # initialization
+        self.left_bottom_position: Tuple[int, int, int]
+        self.left_top_position: Tuple[int, int, int]
+        self.right_top_position: Tuple[int, int, int]
+
+        # motion directions
+        self._a: np.ndarray  # right_top - left_bottom
+        self._b: np.ndarray  # left_top - left_bottom
+        self._c: np.ndarray  # right_top - left_top
+        self._n: np.ndarray
+        self.norm_a: float
+        self.norm_b: float
+        self.norm_c: float
+
+        # eqn of the chip plane
+        self._plane_eqn: np.ndarray
+
+        # current vars
+        self.current_row: Optional[int] = None
+        self.current_pad: Optional[int] = None
+
+        # chip details
+        self.rows: int
+        self.pads: int
+        self.inter_row_dis: float
+        self.inter_pad_dis: float
+
+        # arm kinematics
+        self.speed: int
+        self.acceleration: int
+        self.deceleration: int
+
+        self._arm_pick_up_dis: int
+
+        self._target: np.ndarray
+
+    def set_arm_kinematics(
+        self,
+        speed: int = 100,
+        acceleration: int = 2048,
+        deceleration: int = 2048
+    ) -> None:
+
+        if acceleration % 256 != 0 or deceleration % 256 != 0:
+            raise RuntimeError(
+                "Acceleration and deceleration must be a multiple of 256."
+            )
+
+        self.speed = _convert_micro_meter_to_quadrature_counts(speed)
+        self.acceleration = \
+            _convert_micro_meter_to_quadrature_counts(acceleration)
+        self.deceleration = \
+            _convert_micro_meter_to_quadrature_counts(deceleration)
+
+    def set_pick_up_distance(self, distance: float = 3000) -> None:
+
+        self._arm_pick_up_dis = \
+            _convert_micro_meter_to_quadrature_counts(distance)
+
+    def set_left_bottom_position(self) -> None:
+
+        pos = self.controller.absolute_position()
+        self.left_bottom_position = (pos["A"], pos["B"], pos["C"])
+
+    def set_left_top_position(self) -> None:
+
+        pos = self.controller.absolute_position()
+        self.left_top_position = (pos["A"], pos["B"], pos["C"])
+
+    def set_right_top_position(self) -> None:
+
+        pos = self.controller.absolute_position()
+        self.right_top_position = (pos["A"], pos["B"], pos["C"])
+
+        self._calculate_ortho_vector()
+
+    def _calculate_ortho_vector(self) -> None:
+
+        a = np.array(
+            tuple(
+                map(
+                    lambda i, j: i - j,
+                    self.right_top_position,
+                    self.left_bottom_position,
+                )
+            )
+        )
```

Comment:
shall we let numpy do the vectorized algebra for us ? :) or am i missing something here?
```suggestion
        a = np.asarray(self.right_top_position) - np.asarray(self.left_bottom_position)
```